packages/ember-resolver/lib/core.js

```diff
@@ -57,6 +57,10 @@ define("ember/resolver",
     var prefix, type, name;
     var fullNameParts = fullName.split('@');
 
+    if (/.*:@.*/.test(fullName)) {
```

Comment:
* lets leave a comment here describing why this is needed
* it seems like the above split `fullName.split('@');` (on line 58) is not needed if we hit this branch, this should be cleaned up